src/components/TransactionalInfo.tsx

```diff
@@ -16,35 +16,180 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import styled, { useTheme } from 'styled-components'
+import { calAmountDelta } from '@alephium/sdk'
+import { Output, Transaction } from '@alephium/sdk/api/explorer'
+import { useTranslation } from 'react-i18next'
+import { useParams } from 'react-router-dom'
+import styled from 'styled-components'
 
+import { AddressHash, PendingTx, useAddressesContext } from '../contexts/addresses'
+import { isExplorerTransaction, isPendingTx, TransactionDirection } from '../utils/transactions'
+import AddressBadge from './AddressBadge'
 import Amount from './Amount'
+import Badge from './Badge'
+import DirectionalArrow from './DirectionalArrow'
+import IOList from './IOList'
+import TimeSince from './TimeSince'
+import Token from './Token'
 
 interface TransactionalInfoProps {
-  type: 'out' | 'in' | 'pending'
-  content: string | bigint
+  transaction: Transaction | PendingTx
+  addressHash?: AddressHash
+  hideLabel?: boolean
   className?: string
-  amount?: boolean
-  prefix?: string
 }
 
-const TransactionalInfo = ({ content, className, amount, prefix, type }: TransactionalInfoProps) => {
-  const theme = useTheme()
-  const color = type === 'out' || type === 'pending' ? theme.font.secondary : theme.global.valid
+const TransactionalInfo = ({ transaction: tx, addressHash, className, hideLabel }: TransactionalInfoProps) => {
+  const { addressHash: addressHashParam = '' } = useParams<{ addressHash: AddressHash }>()
+  const _addressHash = addressHash ?? addressHashParam
+
+  const { getAddress } = useAddressesContext()
+  const { t } = useTranslation('App')
+
+  const address = getAddress(_addressHash)
+
+  let amount: bigint | undefined = BigInt(0)
+  let timestamp = 0
+  let type: TransactionDirection
+  let outputs: Output[] = []
+
+  const token = 'alph'
```

Comment:
I see an export of this in `src/components/Token.tsx`. Maybe we can reuse it?

Reply:
That's a type, not a value